lucene/core/src/java/org/apache/lucene/util/QueryBuilder.java

```diff
@@ -63,6 +66,25 @@
   protected boolean enableGraphQueries = true;
   protected boolean autoGenerateMultiTermSynonymsPhraseQuery = false;
 
+  /**
+   * Wraps a term and boost
+   */
+  public static class TermAndBoost {
+    private static final float DEFAULT_BOOST = 1.0f;
```

Comment:
I think this should probably be on `BoostAttribute` rather than here?

Reply:
I agree, coming in the upcoming commit